### PR TITLE
update syntax for empty value labels

### LINKED ISSUE #16526: LABEL com.example.version.is-beta
__docker version:__
```
Client:
 Version:      1.8.2
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   0a8c2e3
 Built:        Thu Sep 10 19:19:00 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.8.2
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   0a8c2e3
 Built:        Thu Sep 10 19:19:00 UTC 2015
 OS/Arch:      linux/amd64
```
__docker info:__
```
Containers: 408
Images: 121
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 937
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.16.0-49-generic
Operating System: Ubuntu 14.04.3 LTS
CPUs: 8
Total Memory: 11.73 GiB
Name: mobile-lab
ID: MNKA:XEBM:UGJ7:HGT4:MN5L:WMCP:ZYIL:7FZZ:K4S2:6Q7I:3K7S:XCTG
WARNING: No swap limit support
```
__uname -a:__
```
Linux mobile-lab 3.16.0-49-generic #65~14.04.1-Ubuntu SMP Wed Sep 9 10:03:23 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```
__Steps to reproduce:__
1. Create a basic dockerfile with only one LABEL layer
```
FROM ubuntu
LABEL com.example.version.is-beta
```
2. Try to build this container

__Results received:__
```
Error response from daemon: Syntax error - can't find = in "com.example.version.is-beta". Must be of the form: name=value
```
__Expected results:__
The container is built correctly.

__Additional info:__
In the official [userguide](https://docs.docker.com/userguide/labels-custom-metadata/), it is mentioned that "The LABEL instruction adds a label to your image, optionally setting its value. ". There are examples in the same context, showing this functionality.

However that is not the case in the actual program, as shown in [line_parsers.go](https://github.com/docker/docker/blob/1ffff4c8e2b208664b6f63ea6cd3c4c3dace3cc9/builder/parser/line_parsers.go#L166)

### PR DESCRIPTION
**Pull request by runcom**
Fix #16526
Discussion #17344

/cc @thaJeztah are there any other docs that should be updated? I haven't found any

Signed-off-by: Antonio Murdaca runcom@redhat.com

### CODE REVIEW

File: docs/userguide/labels-custom-metadata.md [hunk 1]
```diff
@@ -95,14 +95,17 @@ nested properties, the tool itself needs to implement this functionality.
 To add labels to an image, use the `LABEL` instruction in your Dockerfile:
 
 
-    LABEL [<namespace>.]<key>[=<value>] ...
+    LABEL [<namespace>.]<key>=<value> ...
 
-The `LABEL` instruction adds a label to your image, optionally with a value.
+The `LABEL` instruction adds a label to your image. A `LABEL` consists of a `<key>`
+and a `<value>`.
+Use the empty string for labels that don't contain a `<value>`.
```
Comment (thaJeztah) [inline docs/userguide/labels-custom-metadata.md]: Possibly; "Use an empty string for labels that don't have a `<value>`"?

**Comment by thaJeztah** [comment]
Thanks! Looks like the Dockerfile man page still shows `[=<value>]` as optional; https://github.com/docker/docker/blob/master/man/Dockerfile.5.md

**Comment by runcom** [comment]
@thaJeztah updated

**Comment by thaJeztah** [comment]
Awesome! LGTM

**Comment by thaJeztah** [comment]
ping @moxiegirl for review

File: man/Dockerfile.5.md [hunk 2]
```diff
@@ -156,20 +156,23 @@ A Dockerfile is similar to a Makefile.
...
-  -- `LABEL <key>[=<value>] [<key>[=<value>] ...]`or 
+  -- `LABEL <key>=<value> [<key>=<value> ...]`or 
   ```
   LABEL <key>[ <value>]
   LABEL <key>[ <value>]
   ...
   ```
   The **LABEL** instruction adds metadata to an image. A **LABEL** is a
-  key-value pair. To include spaces within a **LABEL** value, use quotes and
+  key-value pair. To specify a **LABEL** without a value, simply use an emtpy
```
Comment (calavera) [inline man/Dockerfile.5.md]: `s/emtpy/empty/`

Comment (thaJeztah) [inline man/Dockerfile.5.md]: :+1: thanks, I missed that one

**Comment by runcom** [comment]
updated, ping @calavera @moxiegirl

**Comment by moxiegirl** [comment]
LGTM

**Comment by moxiegirl** [comment]
@runcom Thank you. Doc build is breaking on another file. I'll sort it out.

**Pull request merged**